syntax = "proto3";

package opencog_services;

message Command {
    string service = 1;
    string cmd = 2;
    int32 session_id = 3;
    repeated string params = 4;
}

message CommandOutput {
    string output = 1;
}

service OpencogServices {
    rpc Execute(Command) returns (CommandOutput) {}
    rpc AsynchronousTask(Command) returns (CommandOutput) {}
}